Remotion/Data/DomainObjects/Infrastructure/ObjectIDConverter.cs

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: (c) RUBICON IT GmbH, www.rubicon.eu
+// SPDX-License-Identifier: LGPL-2.1-or-later
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Infrastructure;
+
+/// <summary> Specialization of <see cref="TypeConverter"/> for conversions from and to <see cref="ObjectID"/>. </summary>
+public class ObjectIDConverter : TypeConverter
+{
+  /// <summary> Test: Is <paramref name="sourceType"/> a <see cref="String"/>? Only <see cref="String"/>s can be converted to <see cref="ObjectID"/>s.</summary>
+  /// <param name="context"> An <see cref="ITypeDescriptorContext"/> that provides a format context. </param>
+  /// <param name="sourceType"> The <see cref="Type"/> of the value to be converted into an <see cref="ObjectID"/> type. </param>
+  /// <returns> <see langword="true"/> if the conversion is supported. </returns>
+  public override bool CanConvertFrom (ITypeDescriptorContext? context, Type sourceType)
+  {
+    ArgumentUtility.CheckNotNull(nameof(sourceType), sourceType);
+
+    return sourceType == typeof(string) || base.CanConvertFrom(context, sourceType);
+  }
+
+  /// <summary> Test: Is <paramref name="destinationType"/> a <see cref="String"/>? <see cref="ObjectID"/>s can only be converted to a <see cref="String"/>s.</summary>
+  /// <param name="context"> An <see cref="ITypeDescriptorContext"/> that provides a format context. </param>
+  /// <param name="destinationType"> The <see cref="Type"/> to convert an <see cref="ObjectID"/> value to. </param>
+  /// <returns> <see langword="true"/> if the conversion is supported. </returns>
+  public override bool CanConvertTo (ITypeDescriptorContext? context, Type? destinationType)
+  {
+    if (destinationType == null)
+      return false;
+
+    return destinationType == typeof(string) || base.CanConvertTo(context, destinationType);
+  }
+
+  /// <summary> Converts <paramref name="value"/> into an <see cref="ObjectID"/> value. </summary>
+  /// <param name="context"> An <see cref="ITypeDescriptorContext"/> that provides a format context. </param>
+  /// <param name="culture"> The <see cref="CultureInfo"/> to use as the current culture. </param>
+  /// <param name="value"> The source value. </param>
+  /// <returns> An <see cref="ObjectID"/> value.  </returns>
+  /// <exception cref="NotSupportedException"> The conversion could not be performed. </exception>
+  public override object? ConvertFrom (ITypeDescriptorContext? context, CultureInfo? culture, object? value)
+  {
+    if (value == null)
+      return null;
+
+    if (!CanConvertFrom(context, value.GetType()))
+      throw new NotSupportedException($"Value of type '{value.GetType()}' cannot be converted to type 'ObjectID'.");
```

Comment:
I don't think we need this anymore? As we now do the base call, it should handle the case where we have an incorrect type.